src/v2i-hub/RSUHealthMonitorPlugin/src/RSUHealthMonitorWorker.cpp

```diff
@@ -0,0 +1,175 @@
+
+            RSUFieldOIDStruct rsuIFMDsrcMsgId = {"rsuIFMDsrcMsgId", RSU_IFM_DSRC_MSG_ID, false};
+            rsuStatusTbl.push_back(rsuIFMDsrcMsgId);
+
+            RSUFieldOIDStruct rsuIFMTxMode = {"rsuIFMTxMode", RSU_IFM_INDEX, false};
+            rsuStatusTbl.push_back(rsuIFMTxMode);
+
+            RSUFieldOIDStruct rsuIFMTxChannel = {"rsuIFMTxChannel", RSU_IFM_TX_CHANNEL, false};
+            rsuStatusTbl.push_back(rsuIFMTxChannel);
+
+            RSUFieldOIDStruct rsuIFMEnable = {"rsuIFMEnable", RSU_IFM_ENABLE, false};
+            rsuStatusTbl.push_back(rsuIFMEnable);
+
+            RSUFieldOIDStruct rsuIFMStatus = {"rsuIFMStatus", RSU_IFM_STATUS, false};
+            rsuStatusTbl.push_back(rsuIFMStatus);
+
+            RSUFieldOIDStruct rsuMode = {"rsuMode", RSU_MODE, true};
+            rsuStatusTbl.push_back(rsuMode);
+
+            RSUFieldOIDStruct rsuChanStatus = {"rsuChanStatus", RSU_CHAN_STATUS, true};
+            rsuStatusTbl.push_back(rsuChanStatus);
+        }
+        return rsuStatusTbl;
+    }
+
+    bool RSUHealthMonitorWorker::isAllRequiredFieldsPresent(const RSUMibVersion &mibVersion, const vector<string> &fields) const
+    {
+        bool isAllPresent = true;
+        for (const auto &config : GetRSUStatusConfig(mibVersion))
+        {
+            if (config.required && find(fields.begin(), fields.end(), config.field) == fields.end())
+            {
+                isAllPresent = false;
+                PLOG(logWARNING) << "No broadcast as required field " << config.field << " is not present!";
+            }
+        }
+        return isAllPresent;
+    }
+
+    RSUStatusConfigTable RSUHealthMonitorWorker::GetRSUStatusConfig(const RSUMibVersion &mibVersion) const
+    {
+        RSUStatusConfigTable result;
+        try
+        {
+            result = _RSUSTATUSConfigMapPtr->at(mibVersion);
+        }
+        catch (const out_of_range &ex)
+        {
+            PLOG(logERROR) << "Unknown MIB version! " << ex.what();
+        }
+        return result;
+    }
+
+    std::map<double, double> RSUHealthMonitorWorker::ParseRSUGPS(const std::string &gps_nmea_data) const
+    {
+        std::map<double, double> result;
+        nmea::NMEAParser parser;
+        nmea::GPSService gps(parser);
+        try
+        {
+            parser.readLine(gps_nmea_data);
+            std::stringstream ss;
+            ss << std::setprecision(8) << std::fixed << gps.fix.latitude << std::endl;
+            auto latitude_str = ss.str();
+            std::stringstream sss;
+            sss << std::setprecision(8) << std::fixed << gps.fix.longitude << std::endl;
+            auto longitude_str = sss.str();
+            result.insert({std::stod(latitude_str), std::stod(longitude_str)});
+            PLOG(logDEBUG) << "Parse GPS NMEA string: " << gps_nmea_data << ". Result (Latitude, Longitude): (" << latitude_str << "," << longitude_str << ")";
+        }
+        catch (nmea::NMEAParseError &e)
```

Comment:
Should be constant

Reply:
Addressed https://github.com/usdot-fhwa-OPS/V2X-Hub/pull/564/commits/a49366ea8030afa8c668ec5879cf8e2d5cf69065